### PR TITLE
(RM) AN(C)OVA: contrasts and simple main effects 

### PR DESCRIPTION
**Pull request by JohnnyDoorn**
- Simple main effects analysis (fix issue #1957)
- Fixed contrasts (fix issue #2198 )
- Added option to pool error term in posthoc test

### CODE REVIEW

File: JASP-Desktop/widgets/tablemodelvariablesoptions.cpp [hunk 1]
```diff
@@ -27,6 +27,7 @@
 #include "options/optionterms.h"
 
 #include "qutils.h"
+#include <QDebug>
```
Comment (vankesteren) [inline JASP-Desktop/widgets/tablemodelvariablesoptions.cpp]: Not necessary, I see no qDebug() statements in the file (right?)

File: JASP-Engine/JASP/R/ancova.R [hunk 2]
```diff
@@ -457,7 +478,7 @@ Ancova <- function(dataset=NULL, options, perform="run", callback=function(...)
 
 	if ( ! is.null(contr))
 		dimnames(contr) <- list(NULL, 1:dim(contr)[2])
-
+  # browser()
```
Comment (vankesteren) [inline JASP-Engine/JASP/R/ancova.R]: Remove :)

File: JASP-Tests/R/tests/testthat/test-ancova.R [hunk 3]
```diff
@@ -68,6 +68,7 @@ test_that("Homogeneity of Variances table results match", {
   expect_equal_tables(table, list(2.72159218177061, 1, 98, 0.102201011380302, 1.57819444559362, 1))
 })
 
+# Contrasts verified with SPSS
```
Comment (vankesteren) [inline JASP-Tests/R/tests/testthat/test-ancova.R]: What's this `SPSS` you speak of? 😛

File: JASP-Engine/JASP/R/ancova.R [hunk 4]
```diff
@@ -1606,6 +1626,143 @@ Ancova <- function(dataset=NULL, options, perform="run", callback=function(...)
...
+    factors[[length(factors)+1]] <- factor
+    lvls[[variable]] <- levels(factor)
+  }
+  if (perform == "run" && status$ready && status$error == FALSE)  {
+  
+    for (level in lvls[[1]]) {
+      subDataset <- subset(dataset, dataset[terms.base64[1]] == level)
+      subOptions <- options
+      subOptions$fixedFactors <- options$fixedFactors[options$fixedFactors != options$moderatorFactorOne]
+      subOptions$modelTerms <- options$modelTerms[ !grepl(options$moderatorFactorOne, unlist(options$modelTerms, recursive = FALSE)) ]
```
Comment (vankesteren) [inline JASP-Engine/JASP/R/ancova.R]: comment here would be nice

File: JASP-Engine/JASP-Engine.pro [hunk 5]
```diff
@@ -75,6 +75,7 @@ INCLUDEPATH += \
 
 linux:INCLUDEPATH += \
 	/usr/share/R/include \
+        /usr/lib/R/library/include \
```
Comment (TimKDJ) [inline JASP-Engine/JASP-Engine.pro]: This is an intended change?

**Pull request merged**